optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/admin/AdminController.java

```diff
@@ -16,36 +16,28 @@
 
 package org.optaweb.employeerostering.service.admin;
 
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.util.Assert;
-import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
 
-import io.swagger.annotations.Api;
-import io.swagger.annotations.ApiOperation;
-
-@RestController
-@RequestMapping("/rest/admin")
-@CrossOrigin
-@Validated
-@Api(tags = "Admin")
+@Path("/rest/admin")
+@ApplicationScoped
+// @Api(tags = "Admin")
 public class AdminController {
 
     private final AdminService adminService;
 
+    @Inject
     public AdminController(AdminService adminService) {
         this.adminService = adminService;
-        Assert.notNull(adminService, "adminService must not be null.");
+        assert adminService != null;// adminService must not be null.
```

Comment:
Use [Objects.requireNonNull()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Objects.html#requireNonNull(T,java.lang.String)). Java assertions are turned off by default: https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html. We never used them.

Reply:
Agree. Java assertions are an obscure JVM feature that is barely ever used by anyone.
First of all because it needs to be enabled to do anything, and if enabled, it just fails the app like any other exception would do.

Reply:
I am debating if we even need this if we are using Injection (which will cause a compile error if there is no implementation). It only good for catching errors within tests basically, but tests also support injection.

Reply:
+1. For `@Inject`ed constructors it is a boiler-plate code with no added value.